R/bainsembayesian.R

```diff
@@ -0,0 +1,150 @@
+#
+# Copyright (C) 2013-2015 University of Amsterdam
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+BainSemBayesian <- function(jaspResults, dataset, options, ...) {
+
+  # What type of Bain analysis is being conducted?
+  type <- "sem"
+
+  # Read the data set
+  dataList <- .bainReadDataset(options, type, dataset)
+
+  # Check if current options allow for analysis
+  ready <- .bainOptionsReady(options, type, dataList[["dataset"]])
+
+  # Check if current data allow for analysis
+  .bainDataReady(dataList[["dataset"]], options, type)
+  
+  # Create a container for the results
+  bainContainer <- .bainGetContainer(jaspResults, deps = c("syntax", "model", "seed", "fraction", "standardized"))
+  
+  # Create a legend containing the order constrained hypotheses
+  .bainLegend(dataList[["dataset"]], options, type, jaspResults, position = 0)
+  
+  # Create a table containing the main analysis results
+  .bainResultsTable(dataList[["dataset"]], options, bainContainer, dataList[["missing"]], ready, type, position = 1)
+  
+  # Create the Bayes factor matrix
+  .bainBfMatrix(dataList[["dataset"]], options, bainContainer, ready, type, position = 2)
+  
+  # Create the descriptive statistics (coefficients) table
+  .bainDescriptivesTable(dataList[["dataset"]], options, bainContainer, ready, type, position = 3)
+  
+  # Create the posterior probability plots
+  .bainPosteriorProbabilityPlot(dataList[["dataset"]], options, bainContainer, ready, type, position = 4)
+  
+  # Create the path diagram plot
+  .bainSemPathDiagram(dataList[["dataset"]], options, bainContainer, ready, jaspResults, position = 5)
+}
+
+.bainLavaanState <- function(dataset, options, bainContainer, ready, jaspResults){
+  if(!is.null(bainContainer[["lavaanResult"]])){
+    return(bainContainer[["lavaanResult"]]$object)
+  } else if(ready){
+    syntax <- .bainSemTranslateModel(options[["syntax"]], dataset)
+    error <- try({
+      fit <- lavaan::sem(model = syntax, data = dataset)
+    })
+    if(isTryError(error)){
+      bainContainer$setError(gettextf("An error occurred in the call to lavaan: %1$s", jaspBase:::.extractErrorMessage(error)))
```

Comment:
Probably doesn't need to be caught, or is something salvageable after this fails?

Reply:
It might still be important to show the specified hypotheses, those can still be displayed